todo/src/actions/TodoActions.js

```diff
@@ -0,0 +1,37 @@
+import { ADD_TODO, TOGGLE_TODO,SHOW_ALL, SHOW_ACTIVE, SHOW_COMPLETED} from '../actions/actionTypes'
+
+let id =0;
+export const addTodo = (todo) => {
+    return {
+        type: ADD_TODO,
+        payload: {
+            id: ++id,
+            todo,
+            completed:false
+        }
+    }
+}
+
+export const toggleTodo = (id) => {
+    return {
+        type: TOGGLE_TODO,
+        id
+    }
+}
+
+
+export const showall = () => {
```

Comment:
You can create a filterAction where the payload decides which type of filter it will have